tests/unit/oidc/test_forms.py

```diff
@@ -288,20 +286,31 @@ def test_validate(self, monkeypatch):
 
         assert form.validate()
 
-    def test_validate_owner(self, monkeypatch):
+    @pytest.mark.parametrize(
+        "repo_slug", ["missing", "missing_repo/" "/missing_user", "", "/excess/slash/"]
+    )
+    def test_validate_repo_slug(self, monkeypatch, repo_slug):
         form = forms.GitHubPublisherForm(api_token=pretend.stub())
 
         owner_info = {"login": "some-username", "id": "1234"}
         monkeypatch.setattr(form, "_lookup_owner", lambda o: owner_info)
 
-        field = pretend.stub(data="SOME-USERNAME")
-        form.validate_owner(field)
+        field = pretend.stub(data=repo_slug)
 
-        assert form.normalized_owner == "some-username"
-        assert form.owner_id == "1234"
+        with pytest.raises(wtforms.validators.ValidationError):
+            form.validate_repo_slug(field)
 
     @pytest.mark.parametrize(
-        "workflow_filename", ["missing_suffix", "/slash", "/many/slashes", "/slash.yml"]
+        "workflow_filename",
+        [
+            "missing_suffix",
+            "/slash",
+            "/many/slashes",
+            "/slash.yml",
+            "repo@/badsym",
+            "user/@badsym",
+            "-user/...",
```

Comment:
I think you meant to put these last three testcases in `test_validate_repo_slug`, not here 🙂

Reply:
(I can move them.)

Reply:
836aa9dac..f3c4090dd